### DISCUSSION TITLE
How can I add additional `lens_mass_model` for seperate light source near the Lens galaxy?

### DISCUSSION THREAD

**Discussion by rafee-adnan** [AstroBridge/BDLensing#146] [Q&A] [answered]
As my lensing system has an additional light source near the lensing system (image below), I was wondering how can I add additional `lens_mass_model` for separate light source near the Lens galaxy? I initially masked out the additional light source, but it is affecting my final gamma value in the MCMC run. 

<img width="428" alt="Screenshot 2023-06-14 at 9 28 19 PM" src="https://github.com/ajshajib/BDLensing/assets/76589056/c1d057e8-59cc-4ad7-a8d9-b78ea306242e">
<img width="430" alt="Screenshot 2023-06-14 at 9 28 52 PM" src="https://github.com/ajshajib/BDLensing/assets/76589056/268e5c0b-1ada-4e17-b3ac-2c4e66baf128">

**Comment by rafee-adnan** [comment]
I am mentioning @numjahan, as @ajshajib vai told me you faced similar situation with your model and added an additional light profile successfully.

**Reply by numjahan** [reply, marked as answer]
I included an `SIE` component in the mass model of the lens galaxy and a `Sersic Ellipse` as the third element in the light model of the lens galaxy.

For adding an additional lens mass model - 

- First, added `SIE` to the `lens_model_list` list.
![image](https://github.com/ajshajib/BDLensing/assets/92924875/623a0f56-ca63-4f2f-9db4-9e8a69d3622c)

- Then, I used the mask's center (the same mask I created for the additional light source for my lensing system previously) as the values for `satellite_center_x` and `satellite_center_y`. Here, `satellite_centroid_bound` is my assumed value for the bound in pixel * arcsecond per pixel. In my case, **satellite_centroid_bound = 5 pixels * 0.08 = 0.4**
![image](https://github.com/ajshajib/BDLensing/assets/92924875/8c222e55-e085-4c9d-bfe6-c1f6b9230922)

- I used the same `center_x` and `center_y` for the 3rd Sersic Ellipse parameters-
![image](https://github.com/ajshajib/BDLensing/assets/92924875/c1a5b780-3da5-4a64-a9ea-f8f559cdbc3a)


- Then, I joined their centers and "e1 & e2" together -
 ![image](https://github.com/ajshajib/BDLensing/assets/92924875/a0962229-f1e6-45d6-b402-e0cfa237d232)

- Then, I added it to the `kwargs_constraints` using the keyword `joint_lens_with_light`.
![image](https://github.com/ajshajib/BDLensing/assets/92924875/1fa13369-80fe-4ac4-b08d-c9b311991581)

**Reply by rafee-adnan** [reply]
I have got a wonderful fit! Thank you! `Reduced Chi-squared value 1.02`
<img width="659" alt="Screenshot 2023-06-16 at 11 26 45 PM" src="https://github.com/ajshajib/BDLensing/assets/76589056/31a2b55e-70b0-4247-931d-4b130b2e6f7f">
